firefox-ios/Ecosia/Analytics/Analytics.swift

```diff
@@ -41,9 +42,14 @@ open class Analytics {
         self.notificationCenter = notificationCenter
     }
 
-    private func track(_ event: SnowplowTracker.Event) {
+    internal func track(_ event: SnowplowTracker.Event) {
         guard User.shared.sendAnonymousUsageData else { return }
+        if let structuredEvent = event as? Structured {
+            appendContextIfNeeded(to: structuredEvent)
+        }
```

Comment:
Is this really supposed to be all events? I see on Journey case it was, but for app events we have the practice to add only to activity (launch and resume). Did you check this with BI?

Reply:
I'm happy you flagged it as a double check. Personally, I obtained the information from the ticket, which specifically instructed Mobile to mimic JOU behavior. I checked the JOU ticket, and it was strictly mentioned to "really, all events".
By making it very clear on their side too, I realised that it must indeed be an exception and we should be doing same.

Reply:
Yeah, that makes sense, but I think it doesn't hurt to double check with them and with the PM since adding this to all events means more data to handle. Native case might always differ a bit from Web since it's not that relevant for revenue tracking.